diff --git a/maemo/main.c b/maemo/main.c
index 3bc7f8f..dcd7fd7 100644
--- a/maemo/main.c
+++ b/maemo/main.c
@@ -176,6 +176,7 @@ int main(int argc, char **argv)
 	ChangeWorkingDirectory("c");
 	char file[MAXPATHLEN] = "";
 	char path[MAXPATHLEN];
+	char isofilename[MAXPATHLEN] = "";
 	const char *cdfile = NULL;
 	int loadst = 0;
 	int i;
@@ -207,7 +208,6 @@ int main(int argc, char **argv)
 		     if (!strcmp(argv[i], "-psxout")) Config.PsxOut = 1;
 		else if (!strcmp(argv[i], "-load")) loadst = atol(argv[++i]);
 		else if (!strcmp(argv[i], "-cdfile")) {
-			char isofilename[MAXPATHLEN];
 			if (i+1 >= argc) break;
 			strncpy(isofilename, argv[++i], MAXPATHLEN);
 			if (isofilename[0] != '/') {
