--- a/overrides/mame2000.h
+++ b/overrides/mame2000.h
@@ -4,7 +4,7 @@ static const struct core_override_option mame2000_core_option_overrides[] = {
 	{
 		.key = "mame2000-frameskip",
 		.info = "Skip frames to avoid audio crackling. Improves performance at the expense of visual smoothness.",
-		.default_value = "auto",
+		.default_value = "disabled",
 		.retro_var_value = "Frameskip ; disabled|auto|threshold"
 	},
 	{
